sdk/src/client/api/block_builder/input_selection/core/mod.rs

```diff
@@ -403,10 +403,97 @@ impl InputSelection {
+                }
+                _ => {}
+            }
+        }
+
+        // Validate alias output transition
+        for output in self.outputs.iter() {
+            match output {
+                Output::Alias(alias_output) => {
+                    // Null id outputs are just minted and can't be a transition
+                    if alias_output.alias_id().is_null() {
+                        continue;
+                    }
+
+                    let alias_input = input_aliases
+                        .iter()
+                        .find(|i| {
+                            if let Output::Alias(alias_input) = &i.output {
+                                *alias_output.alias_id() == alias_input.alias_id_non_null(i.output_id())
+                            } else {
+                                false
+                            }
+                        })
+                        .expect("ISA is broken because there is no alias input");
+
+                    if let Err(err) = AliasOutput::transition_inner(
+                        alias_input.output.as_alias(),
+                        alias_output,
+                        &input_chains_foundries,
+                        &self.outputs,
+                    ) {
+                        log::debug!("validate_transitions error {err:?}");
+                        let alias_transition =
+                            if alias_input.output.as_alias().state_index() == alias_output.state_index() {
+                                AliasTransition::Governance
+                            } else {
+                                AliasTransition::State
+                            };
+                        return Err(Error::UnfulfillableRequirement(Requirement::Alias(
+                            *alias_output.alias_id(),
+                            alias_transition,
+                        )));
+                    }
+                }
+                Output::Nft(nft_output) => {
+                    // Null id outputs are just minted and can't be a transition
+                    if nft_output.nft_id().is_null() {
+                        continue;
+                    }
+
+                    let nft_input = input_nfts
+                        .iter()
+                        .find(|i| {
+                            if let Output::Nft(nft_input) = &i.output {
+                                *nft_output.nft_id() == nft_input.nft_id_non_null(i.output_id())
+                            } else {
+                                false
+                            }
+                        })
+                        .expect("ISA is broken because there is no nft input");
+
+                    if nft_input.output.as_nft().immutable_features() != nft_output.immutable_features() {
+                        return Err(Error::UnfulfillableRequirement(Requirement::Nft(*nft_output.nft_id())));
+                    }
+                }
+                _ => {}
```

Comment:
What about foundry transitions?

Reply:
Thought about doing it in a separate PR, but it's actually not that much, will also do it